### PR TITLE
Implement copying/moving constructors of InputAqlItemRow

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Improve performance decline of AQL sort through #10025 again.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6588/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/Performance/job/arangodb-simple-performance--ANY-linux/372/

File: arangod/Aql/InputAqlItemRow.cpp [hunk 1]
```diff
@@ -254,7 +254,12 @@ void InputAqlItemRow::toVelocyPack(transaction::Methods* trx, VPackBuilder& resu
 InputAqlItemRow::InputAqlItemRow(CreateInvalidInputRowHint)
     : _block(nullptr), _baseIndex(0) {}
 
-InputAqlItemRow::InputAqlItemRow(SharedAqlItemBlockPtr block, size_t baseIndex)
+InputAqlItemRow::InputAqlItemRow(SharedAqlItemBlockPtr const& block, size_t baseIndex)
+    : _block(block), _baseIndex(baseIndex) {
+  TRI_ASSERT(_block != nullptr);
+}
+
+InputAqlItemRow::InputAqlItemRow(SharedAqlItemBlockPtr&& block, size_t baseIndex)
```
Comment (graetzer) [inline arangod/Aql/InputAqlItemRow.cpp]: shouldn't this be noexcept

Comment (jsteemann) [reply arangod/Aql/InputAqlItemRow.cpp]: :+1:

Comment (goedderz) [reply arangod/Aql/InputAqlItemRow.cpp]: Yes!

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6594/

**Pull request merged**